ui/ducks/send/swap-and-send-utils.ts

```diff
@@ -0,0 +1,198 @@
+import { isValidHexAddress } from '../../../shared/modules/hexstring-utils';
+
+type ADDRESS = `0x${string}`;
+
+type Request = {
+  chainId: number; // chain ID as decimal
+  sourceAmount: string; // big number string
+  sourceToken: ADDRESS;
+  destinationToken: ADDRESS;
+  sender: ADDRESS;
+  recipient: ADDRESS;
+  slippage: string; // slippage as a percentage; e.g. '1.0' for 1%
+};
+
+type Quote = {
+  gasParams: {
+    maxGas: number;
+  };
+  trade: {
+    data: string;
+    to: string;
+    from: string;
+    value: string;
+  };
+  approvalNeeded: null | {
+    data: string;
+    to: string;
+    from: string;
+  };
+  sourceAmount: string;
+  destinationAmount: string;
+  sourceToken: string;
+  destinationToken: string;
+  sender: string;
+  recipient: string;
+  aggregator: string;
+  aggregatorType: string;
+  error: null | object;
+  fee: number;
+};
+
+const QUOTE_VALIDATORS = [
+  {
+    property: 'gasParams',
+    type: 'object',
+    validator: (gasParams: Record<string, any>) =>
+      gasParams && isNumber(gasParams.maxGas),
+  },
+  {
+    property: 'trade',
+    type: 'object',
+    validator: (trade: Record<string, any>) =>
+      trade &&
+      validHex(trade.data) &&
+      isValidHexAddress(trade.to, { allowNonPrefixed: false }) &&
+      isValidHexAddress(trade.from, { allowNonPrefixed: false }) &&
+      truthyString(trade.value),
+  },
+  {
+    property: 'approvalNeeded',
+    type: 'object',
+    validator: (approvalTx: Record<string, any>) =>
+      approvalTx === null ||
+      (approvalTx &&
+        validHex(approvalTx.data) &&
+        isValidHexAddress(approvalTx.to, { allowNonPrefixed: false }) &&
+        isValidHexAddress(approvalTx.from, { allowNonPrefixed: false })),
+  },
+  {
+    property: 'sourceAmount',
+    type: 'string',
+    validator: truthyDigitString,
+  },
+  {
+    property: 'destinationAmount',
+    type: 'string',
+    validator: truthyDigitString,
+  },
+  {
+    property: 'sourceToken',
+    type: 'string',
+    validator: (input: string) =>
+      isValidHexAddress(input, { allowNonPrefixed: false }),
+  },
+  {
+    property: 'destinationToken',
+    type: 'string',
+    validator: (input: string) =>
+      isValidHexAddress(input, { allowNonPrefixed: false }),
+  },
+  {
+    property: 'sender',
+    type: 'string',
+    validator: (input: string) =>
+      isValidHexAddress(input, { allowNonPrefixed: false }),
+  },
+  {
+    property: 'recipient',
+    type: 'string',
+    validator: (input: string) =>
+      isValidHexAddress(input, { allowNonPrefixed: false }),
+  },
+  {
+    property: 'aggregator',
+    type: 'string',
+    validator: truthyString,
+  },
+  {
+    property: 'aggregatorType',
+    type: 'string',
+    validator: truthyString,
+  },
+  {
+    property: 'error',
+    type: 'object',
+    validator: (error: any) => error === null || typeof error === 'object',
+  },
+  {
+    property: 'fee',
+    type: 'number',
+  },
+];
```

Comment:
Can you add one for `priceSlippage` as well?

Reply:
https://swap.metaswap.codefi.network/v2/networks/1/quotes?sourceAmount=30000000000000&sourceToken=0x0000000000000000000000000000000000000000&destinationToken=0xdac17f958d2ee523a2206206994597c13d831ec7&sender=0x676759c59b412e93e741c84db81c8d7c18ce4845&recipient=0x0a985A957B490F4D05beF05bC7eC556Dd8535946&slippage=3

I'm seeing a bunch of `null` for that property. If we don't use it I'm in favor of leaving it out.